backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandra.java

```diff
@@ -28,12 +31,110 @@
+     *
+     * This process is done by using the default user provided by docker cassandra, it has the capability of creating roles,
+     * keyspaces, and granting permissions to those entities.
+     */
+    public static class CassandraResourcesManager {
+
+        private static final String CASSANDRA_SUPER_USER = "cassandra";
+        private static final String CASSANDRA_SUPER_USER_PASSWORD = "cassandra";
+
+        private final DockerCassandra cassandra;
+
+        private CassandraResourcesManager(DockerCassandra cassandra) {
+            this.cassandra = cassandra;
+        }
+
+        public void initializeKeyspace(String keyspace) {
+            ClusterConfiguration configuration = configurationBuilder(keyspace).build();
+
+            try (Cluster privilegedCluster = ClusterFactory.create(configuration)) {
+                provisionNonPrivilegedUser(privilegedCluster);
+                KeyspaceFactory.createKeyspace(configuration, privilegedCluster);
+                grantPermissionTestingUser(privilegedCluster, keyspace);
+            }
+        }
+
+        public void dropKeyspace(String keyspace) {
+            try (Cluster cluster = ClusterFactory.create(
+                configurationBuilder(keyspace).build())) {
+                try (Session cassandraSession = cluster.newSession()) {
+                    boolean applied = cassandraSession.execute(
+                        SchemaBuilder.dropKeyspace(keyspace)
+                            .ifExists())
+                        .wasApplied();
+
+                    if (!applied) {
+                        throw new IllegalStateException("cannot drop keyspace '" + keyspace + "'");
+                    }
+                }
+            }
+
+        }
+
+        public boolean keyspaceExist(String keyspace) {
+            try (Cluster cluster = ClusterFactory.create(
+                configurationBuilder(keyspace).build())) {
+                return KeyspaceFactory.keyspaceExist(cluster, keyspace);
+            }
+        }
+
+        private void provisionNonPrivilegedUser(Cluster privilegedCluster) {
+            try (Session session = privilegedCluster.newSession()) {
+                session.execute("CREATE ROLE IF NOT EXISTS "+ CASSANDRA_TESTING_USER + " WITH PASSWORD = '" + CASSANDRA_TESTING_PASSWORD + "' AND LOGIN = true");
+            }
+        }
+
+        private void grantPermissionTestingUser(Cluster privilegedCluster, String keyspace) {
+            try (Session session = privilegedCluster.newSession()) {
+                session.execute("GRANT CREATE ON KEYSPACE " + keyspace + " TO " + CASSANDRA_TESTING_USER);
+                session.execute("GRANT SELECT ON KEYSPACE " + keyspace + " TO " + CASSANDRA_TESTING_USER);
+                session.execute("GRANT MODIFY ON KEYSPACE " + keyspace + " TO " + CASSANDRA_TESTING_USER);
+                session.execute("GRANT DROP ON KEYSPACE "+ keyspace + " TO " + CASSANDRA_TESTING_USER); // some tests require dropping in setups
```

Comment:
no protection

Reply:
I don't get it

Reply:
not preparing queries is dangerous, even in tests, we will loose a lot of time during debugging

Reply:
I'm sorry I tried, but cassandra refuse my prepared statement `com.datastax.driver.core.exceptions.SyntaxError: Bind variables cannot be used for role names`